python/pyspark/worker.py

```diff
@@ -295,7 +295,8 @@ def read_udfs(pickleSer, infile, eval_type):
     is_map_iter = eval_type == PythonEvalType.SQL_MAP_PANDAS_ITER_UDF
 
     if is_scalar_iter or is_map_iter:
-        assert num_udfs == 1, "One SCALAR_ITER UDF expected here."
+        assert is_scalar_iter and num_udfs == 1, "One SCALAR_ITER UDF expected here."
+        assert is_map_iter and num_udfs == 1, "One MAP_ITER UDF expected here."
```

Comment:
Can the two `assert` both be passed? E.g., if `is_scalar_iter`, then another `assert` will fail?

Reply:
Yea let me clean up.